proto/public/uber/cadence/api/v1/common.proto

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package uber.cadence.api.v1;
+
+option go_package = "apiv1";
+option java_multiple_files = true;
+option java_outer_classname = "ApiProto";
+option java_package = "com.uber.cadence.api.v1";
+
+import "google/protobuf/duration.proto";
+import "dependencies/gogoproto.proto";
+
+message WorkflowExecution {
+  string workflow_id = 1;
+  string run_id = 2;
+}
+
+message WorkflowType {
+  string name = 1;
+}
+
+message ActivityType {
+  string name = 1;
+}
+
+message Payload {
+  bytes data = 1;
+}
```

Comment:
I kinda wonder if it'd be worth replacing all use of `Payload` with `DataBlob`, with an `ENCODING_TYPE_UNKNOWN / _OPAQUE`... might make it possible to "upgrade" things in-place later.

e.g. `ScheduleActivityTaskDecisionAttributes` -> if `_PROTO3`, parse the `data` as an `Any`, and you might be able to understand and display it in a UI.  Then the unknown/opaque behavior is just like it is now: try to display, maybe it's json, maybe not.

Reply:
hmms.  on more thinking: this could also provide a way to point to external resources (encoding_type_uri -> schema://path/to/resource) for large blobs.  which could be valid in _many_ uses of Payload, though not all (e.g. it'd be kinda crazy for `SearchAttributes`, but it'd be perfect for `RespondActivityTaskCompletedRequest`)

Reply:
Introduction of `Payload` wrapper was exactly for this purpose, to add encoding type (or maybe other metadata) next to the actual data later, if needed. Now, your suggestion to reuse the same DataBlob type, actually makes a lot of sense. I'm considering doing it.

Reply:
yea.  if there's a clear separation between payload-purposes and datablob-purposes, maybe it makes sense to separate them.  otherwise "a binary blob" seems pretty generic to me, and runtime limits on what's allowed in what API seem fine (similar to size limits.  not necessarily worth ensuring it via the type system)